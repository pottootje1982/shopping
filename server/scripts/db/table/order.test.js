describe('storeRecipe()', () => {
  let orderDb, recipeDb

  beforeAll(async () => {
    ;({ orderDb, recipeDb } = global)
  })

  it('get orders', async () => {
    const orders = await orderDb.get()
    expect(orders.length).toEqual(0)
  })

  it('get hydrated', async () => {
    const recipes = await recipeDb.getRecipes(undefined, 'ah')
    let order = [
      {
        uid: '3fe04f98-8d73-4e9d-a7da-f4c1241aa3c4',
        parsedIngredients: [
          {
            ingredient: 'prei',
            product: {
              id: 171425,
              quantity: 1
            }
          },
          {
            ingredient: 'aardappels',
            product: {
              quantity: 10
            }
          }
        ]
      }
    ]
    const user = 'henry jones'
    await orderDb.storeOrder(order, 'ah', user)
    const orders = await orderDb.getHydrated(user, recipes)
    expect(orders.length).toBe(1)
    order = orders[0]
    expect(order.date).toBeDefined()
    expect(order.recipes.length).toBe(1)
    const recipe = order.recipes[0]

    expect(recipe.name).toEqual('Zalm met prei')
    expect(recipe.parsedIngredients.length).toBe(8)
    expect(recipe.parsedIngredients).toMatchSnapshot()

    await orderDb.remove({ date: order.date })
  })

  it('store order', async () => {
    let order = [
      {
        uid: '5134b9ac-32fd-4e5c-a6da-681d33cd007f',
        parsedIngredients: [
          {
            ingredient: 'slagroom',
            product: {
              id: 191621,
              quantity: 1
            }
          },
          {
            ingredient: 'dille',
            product: {
              id: 238966,
              quantity: 1
            }
          },
          {
            ingredient: 'witte wijn',
            product: {
              id: 183474,
              quantity: 1
            }
          },
          {
            ingredient: 'boter',
            product: {
              id: 58082,
              quantity: 1
            }
          },
          {
            ingredient: 'sjalotjes',
            product: {
              id: 160653,
              quantity: 2
            }
          },
          {
            ingredient: 'boontjes',
            product: {
              id: 4102,
              quantity: 2
            }
          },
          {
            ingredient: 'witvis',
            product: { notAvailable: true, ignore: false, quantity: 1 }
          },
          {
            ingredient: 'aardappels',
            product: {
              id: 111388,
              quantity: 1
            }
          }
        ]
      }
    ]
    await orderDb.storeOrder(order)
    const orders = await orderDb.get()
    expect(orders.length).toEqual(1)
    order = orders[0]
    expect(order.recipes.length).toEqual(1)
    const recipe = order.recipes[0]
    expect(recipe.parsedIngredients).toMatchSnapshot()

    await orderDb.remove({ uid: order.uid })
  })
})
