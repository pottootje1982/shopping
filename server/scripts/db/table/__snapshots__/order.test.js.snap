// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storeRecipe() get hydrated 1`] = `
Ingredients [
  Object {
    "all": Array [
      undefined,
      undefined,
      "Prei",
    ],
    "full": "Prei",
    "ingredient": "Prei",
    "product": Object {
      "id": 171425,
      "quantity": 1,
      "title": "AH Prei",
    },
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      undefined,
      undefined,
      "Dille",
    ],
    "full": "Dille",
    "ingredient": "Dille",
    "product": Object {
      "id": 238966,
      "title": "AH Dille",
    },
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      undefined,
      "Blik",
      "tomaten",
    ],
    "full": "Blik tomaten",
    "ingredient": "tomaten",
    "product": Object {
      "id": 395307,
      "title": "AH Biologisch Tomatenblokjes",
    },
    "quantity": 1,
    "unit": "Blik",
  },
  Object {
    "all": Array [
      undefined,
      undefined,
      "Zalm",
    ],
    "full": "Zalm",
    "ingredient": "Zalm",
    "product": Object {
      "id": 51335,
      "ignore": true,
      "title": "AH Zalmfilet",
    },
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      3,
      undefined,
      "Aardappels",
    ],
    "full": "3 Aardappels",
    "ingredient": "Aardappels",
    "product": Object {
      "id": 111388,
      "quantity": 10,
      "title": "AH Kilo'tje iets kruimige aardappelen",
    },
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      undefined,
      undefined,
      "Wijn",
    ],
    "full": "Wijn",
    "ingredient": "Wijn",
    "product": Object {},
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      undefined,
      undefined,
      "Honing",
    ],
    "full": "Honing",
    "ingredient": "Honing",
    "product": Object {},
    "quantity": 1,
    "unit": undefined,
  },
  Object {
    "all": Array [
      undefined,
      undefined,
      "koriander poeder",
    ],
    "full": "koriander poeder",
    "ingredient": "koriander poeder",
    "product": Object {},
    "quantity": 1,
    "unit": undefined,
  },
]
`;

exports[`storeRecipe() store order 1`] = `
Array [
  Object {
    "ingredient": "slagroom",
    "product": Object {
      "id": 191621,
      "quantity": 1,
    },
  },
  Object {
    "ingredient": "dille",
    "product": Object {
      "id": 238966,
      "quantity": 1,
    },
  },
  Object {
    "ingredient": "witte wijn",
    "product": Object {
      "id": 183474,
      "quantity": 1,
    },
  },
  Object {
    "ingredient": "boter",
    "product": Object {
      "id": 58082,
      "quantity": 1,
    },
  },
  Object {
    "ingredient": "sjalotjes",
    "product": Object {
      "id": 160653,
      "quantity": 2,
    },
  },
  Object {
    "ingredient": "boontjes",
    "product": Object {
      "id": 4102,
      "quantity": 2,
    },
  },
  Object {
    "ingredient": "witvis",
    "product": Object {
      "ignore": false,
      "notAvailable": true,
      "quantity": 1,
    },
  },
  Object {
    "ingredient": "aardappels",
    "product": Object {
      "id": 111388,
      "quantity": 1,
    },
  },
]
`;
